pkg/cli/cli.go

```diff
@@ -983,14 +1001,99 @@ func printConnection(c *command, cvd RemoteCVDLocator, status ConnStatus) {
 	c.Printf("%s/%s: %s\n", cvd.Host, cvd.WebRTCDeviceID, state)
 }
 
-func buildAgentCmdArgs(flags *ConnectFlags, device string) []string {
-	args := []string{
-		ConnectionAgentCommandName,
-		device,
-	}
+func buildAgentCmdArgs(flags *ConnectFlags, device string, connectAgent string) []string {
+	args := []string{connectAgent, device}
 	return append(args, flags.AsArgs()...)
 }
 
+func runConnectionProxyAgentCommand(flags *ConnectFlags, c *command, args []string, opts *subCommandOpts) error {
+	if len(args) > 1 {
+		return fmt.Errorf("connection agent only supports a single device, received: %v", args)
+	}
+	if len(args) == 0 {
+		return errors.New("missing device")
+	}
+	device := args[0]
+	service, err := opts.ServiceBuilder(flags.CVDRemoteFlags, c.Command)
+	if err != nil {
+		return err
+	}
+	hosts, err := service.ListHosts()
+	if err != nil {
+		return fmt.Errorf("error listing hosts: %w", err)
+	}
+	var host *apiv1.HostInstance
+	for _, instance := range hosts.Items {
+		if instance.Name == flags.host {
+			host = instance
+		}
+	}
+	if host == nil {
+		return errors.New("failed to find host")
+	}
+	if host.Docker == nil {
+		return errors.New("instance type should be Docker")
+	}
+
+	controlDir := opts.InitialConfig.ConnectionControlDirExpanded()
+	cvdHosts, err := listCVDsSingleHost(service, controlDir, flags.host)
+	if err != nil {
+		return fmt.Errorf("error listing CVDs: %w", err)
+	}
+	var port string
+	for _, cvd := range cvdHosts[0].CVDs {
```

Comment:
to improve readability, let's split this logic into two parts:

1. find cvd
2. get port value.

something like.
```
cvd := findCVD(host, name)
_, port, err := net.SplitHostPort(cvd.ADBSerial)
```

Reply:
Done.